hkmc2/shared/src/main/scala/hkmc2/codegen/llir/Builder.scala

```diff
@@ -202,72 +383,102 @@ final class LlirBuilder(tl: TraceLogger)(fresh: Fresh, fnUid: FreshInt, clsUid:
       case Call(Value.Ref(sym: BuiltinSymbol), args) =>
         bArgs(args):
           case args: Ls[TrivialExpr] =>
-            val v = fresh.make
-            Node.LetExpr(v, Expr.BasicOp(sym.nme, args), k(v |> sr))
-      case Call(Value.Ref(sym), args) if sym.nme.head.isUpper =>
+            val v: Local = newTemp
+            Node.LetExpr(v, Expr.BasicOp(sym, args), k(v |> sr))
+      case Call(Value.Ref(sym), args) if sym.nme.isCapitalized =>
```

Comment:
There is no longer a need for this hack (determining if something is a class by its name's capitalization).